src/ac_training_lab/opentrons/ot-2/_scripts/OT2mqtt.py

```diff
@@ -0,0 +1,184 @@
+
+with open("ac_6_tuberack_15000ul.json") as labware_file2:
+    labware_def2 = json.load(labware_file2)
+    reservoir = protocol.load_labware_from_definition(labware_def2, 3)
+
+plate = protocol.load_labware(load_name="corning_96_wellplate_360ul_flat", location=1)
+
+tiprack_1 = protocol.load_labware(load_name="opentrons_96_tiprack_300ul", location=9)
+
+p300 = protocol.load_instrument(
+    instrument_name="p300_single_gen2", mount="right", tip_racks=[tiprack_1]
+)
+
+p300.well_bottom_clearance.dispense = 8
+
+
+print("Labwares loaded")
+
+
+def mix_color(payload):
+    R = payload["command"]["R"]
+    Y = payload["command"]["Y"]
+    B = payload["command"]["B"]
+    mix_well = payload["command"]["well"]
+    session_id = payload["session_id"]
+    experiment_id = payload["experiment_id"]
+
+    total = int(R + Y + B)
+    if total <= 300:
+        raise ValueError("The sum of the proportions must be smaller than 300")
+
+    position = ["B1", "B2", "B3"]
+    portion = {"B1": R, "B2": Y, "B3": B}
+    # total_volume = 280
+    red_volume = R  # int(portion["B1"] * total_volume)
+    yellow_volume = Y  # int(portion["B2"] * total_volume)
+    blue_volume = B  # int(portion["B3"] * total_volume)
+    color_volume = {"B1": red_volume, "B2": yellow_volume, "B3": blue_volume}
+
+    # To DO: start the mixing from the color with the smallest volume
+
+    for pos in position:
+        if float(portion[pos]) != 0.0:
+            p300.pick_up_tip(tiprack_1[pos])
+            p300.aspirate(color_volume[pos], reservoir[pos])
+            p300.dispense(color_volume[pos], plate[mix_well])
+            p300.default_speed = 100  # reduce pipette speed
+            p300.blow_out(reservoir["A1"].top(z=-5))
+            p300.default_speed = 400  # reset pipette speed
+            p300.drop_tip(tiprack_1[pos])
+
+    p300.pick_up_tip(tiprack_2["A2"])
+    p300.move_to(plate[mix_well].top(z=3))
+
+    # payloadtosent = json.dumps(payload)
+    # print("Sending read command to sensor...")
+    # client.publish(SENSOR_COMMAND_TOPIC, payloadtosent, qos=2)
+
+    print("Sending status to HF...")
+
+    payload = (
+        f'{{"status": {{"sensor_status":"in_place"}}, '
+        f'"experiment_id": "{experiment_id}", '
+        f'"session_id": "{session_id}"}}'
+    )
+
+    client.publish(OT2_STATUS_TOPIC, payload, qos=2)  # send a status message back to HF
+
+
+def move_sensor_back(payload):
+    results_status = payload["command"]["sensor_status"]
+    session_id = payload["session_id"]
+    experiment_id = payload["experiment_id"]
+
+    p300.drop_tip(tiprack_2["A2"].top(z=-80))
+    # protocol.home()
+
+    payload = (
+        f'{{"status": {{"sensor_status":"charging"}}, '
+        f'"experiment_id": "{experiment_id}", '
+        f'"session_id": "{session_id}"}}'
+    )
+    client.publish(OT2_STATUS_TOPIC, payload, qos=2)  # send a status message back to HF
+
+    if results_status == "sensor_timeout":
```

Comment:
I guess we need to think about what happens if we get a sensor timeout from a user perspective